src/common.js

```diff
@@ -17,7 +17,10 @@ export const checkStatus = (response) => {
 
   let error = new Error(response.statusText);
   error.response = response;
-  throw error;
+
+  return response.json()
+    .then(json => Promise.reject(Object.assign(error, json)))
```

Comment:
unless i'm misreading this: isn't this `Promise.reject` just going to immediately trigger the `.catch` on the line after it? Then in that `.catch`, since the promise is rejected with `error` it wouldn't include the json fields that were parsed out in this line.